pytest_mergify/flaky_detection.py

```diff
@@ -140,31 +150,39 @@ def detect_from_report(self, report: _pytest.reports.TestReport) -> bool:
             return False
 
         test = report.nodeid
-        if test in self._context.existing_test_names:
-            return False
+
+        if self.mode == FlakyDetectorMode.NEW:
+            if test in self._context.existing_test_names:
+                return False
+        elif self.mode == FlakyDetectorMode.UNHEALTHY:
+            if test not in self._context.unhealthy_test_names:
+                return False
```

Comment:
```suggestion
        if self.mode == FlakyDetectorMode.NEW and test in self._context.existing_test_names:
                return False
        elif self.mode == FlakyDetectorMode.UNHEALTHY and test not in self._context.unhealthy_test_names:
                return False
```
nitpick, just to avoid nesting the statements, wdyt ?